instrumentation/scala-2.13.0/src/main/scala/com/nr/agent/instrumentation/scala/FutureOps.scala

```diff
@@ -0,0 +1,52 @@
+package com.nr.agent.instrumentation.scala
+
+import com.newrelic.api.agent.{NewRelic, Trace, Transaction}
+
+case class TransactionWrapper(val txn: Transaction) extends AnyVal;
+
+object FutureOps {
+  // making the implicit txnWrapper a `def` ensures getTransaction is envoked each time
+  // a txnWrapper instance is needed for traceFunc/traceBody
+  implicit def txnWrapper = {
+    printTxnInfo()
+    TransactionWrapper(NewRelic.getAgent().getTransaction())
+  }
+
+  // Initial utility to check transaction being shared across threads
+  // TODO: remove once tests assert Txn shared correctly across threads
+  def printTxnInfo(msg: String = "") = {
+    val txn = NewRelic.getAgent().getTransaction()
+    println(s"[${Thread.currentThread.getName}]: txn, $txn, nameSet: ${txn.isTransactionNameSet}, message {$msg}")
+  }
+
+  def traceBody[S](segmentName: String)(body: => S)(implicit txnWrapper: TransactionWrapper): S = {
```

Comment:
Not sure if this is still a WIP, or if was waiting for a review.

I think this is not in the right place. It looks like these are supposed to be used by the costumers in their code.
If it is in this module, it will be weaved into the class at runtime.
So it will not be available to the customer's code.
To be accessible to the customers it should go into the api module.

Reply:
Hey, just moved out of WIP. API has been moved to a separate sub project